extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/BasicWebSocketConnectorImpl.java

```diff
@@ -239,7 +239,11 @@ public void handle(Void event) {
                         trafficLogger.connectionClosed(connection);
                     }
                     if (closeHandler != null) {
-                        doExecute(connection, new CloseReason(ws.closeStatusCode(), ws.closeReason()), closeHandler);
+                        int code = CloseReason.INTERNAL_SERVER_ERROR.getCode();
```

Comment:
Did you actually observe a NPE in your app? Because the javadoc `WebSocket.closeHandler(Handler<Void>)` is pretty clear that _"When the WebSocket received a close frame, the `closeStatusCode()` will return the status code and `closeReason()` will return the reason."_

But maybe I just misundertood.

We could reuse the `CloseReason.INTERNAL_SERVER_ERROR` constant, i.e. something like:
```java
CloseReason reason = CloseReason.INTERNAL_SERVER_ERROR;
if (ws.closeStatusCode() != null) {
   reason = new CloseReason(ws.closeStatusCode(), ws.closeReason());
}
doExecute(connection, reason, closeHandler);
```
WDYT?

Reply:
> Did you actually observe a NPE in your app? Because the javadoc `WebSocket.closeHandler(Handler<Void>)` is pretty clear that _"When the WebSocket received a close frame, the `closeStatusCode()` will return the status code and `closeReason()` will return the reason."_

Yes for example closing a websocket server [websocat](https://github.com/vi/websocat) with CTRL+C.
This is also being done in 
https://github.com/quarkusio/quarkus/blob/57258867fba1ca9f07ec3a639292e70fd1cecc57/extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectionBase.java#L154